addons/point_of_sale/tests/test_frontend.py

```diff
@@ -1344,6 +1344,45 @@ def test_combo_with_custom_attribute(self):
         self.main_pos_config.with_user(self.pos_user).open_ui()
         self.start_tour(f"/pos/ui?config_id={self.main_pos_config.id}", 'test_combo_with_custom_attribute', login="pos_user")
 
+    def test_invoice_due_date(self):
+        """"
+        Tests that the invoice due date is set like it should be regarding the payment term of the partner
+        """
+        payment_term = self.env['account.payment.term'].create({
+            'name': "Payment Term",
+            'line_ids': [Command.create({
+                'value': 'percent',
+                'nb_days': 10,
+                'value_amount': 100,
+            })]
+        })
+        self.partner_test_1.property_payment_term_id = payment_term.id
+        self.env['product.product'].create({
+            'name': 'Test Product',
+            'available_in_pos': True,
+            'list_price': 1000,
+        })
+        self.main_pos_config.open_ui()
+        self.start_tour("/pos/ui?config_id=%d" % self.main_pos_config.id, 'test_invoice_due_date', login="accountman")
+
+        order_with_term = self.env['pos.order'].search([('partner_id', '=', self.partner_test_1.id)], limit=1)
+        self.assertTrue(order_with_term)
+        self.assertEqual(order_with_term.partner_id.id, self.partner_test_1.id)
+
+        invoice = self.env['account.move'].search([('invoice_origin', '=', order_with_term.name)], limit=1)
+        self.assertTrue(invoice)
+        self.assertEqual(invoice.invoice_date + timedelta(days=10), invoice.invoice_date_due)
```

Comment:
```suggestion
        order_with_term = self.env['pos.order'].search([('partner_id', '=', self.partner_test_1.id)], limit=1)
        invoice = self.env['account.move'].search([('invoice_origin', '=', order_with_term.name)], limit=1)
        self.assertEqual(invoice.invoice_date + timedelta(days=10), invoice.invoice_date_due)
```
This is enough, same applies for the code below.